src/main/java/org/opentripplanner/middleware/tripmonitor/TrustedCompanion.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    /**
+     * Extract the accept key from the request parameters.
+     */
+    private static String getAcceptKeyFromRequest(Request request) throws IllegalArgumentException {
+        // Note: optional is true so a missing accept key will be handled here.
+        String acceptKey = HttpUtils.getQueryParamFromRequest(request, ACCEPT_KEY, true);
+        if (acceptKey == null || acceptKey.isEmpty()) {
+            throw new IllegalArgumentException("Accept key not provided.");
+        }
+        return acceptKey;
+    }
+
+    /**
+     * Retrieve the dependent user matching the accept key.
+     */
+    private static OtpUser getUserFromAcceptKey(String acceptKey) throws IllegalArgumentException {
+        if (acceptKey == null) {
+            return null;
+        }
+        OtpUser user = getUserForAcceptKey(acceptKey);
+        if (user == null) {
+            throw new IllegalArgumentException("Otp user unknown.");
+        }
+        return user;
+    }
+
+    public static void manageAcceptDependentEmail(OtpUser dependentUser) {
+        manageAcceptDependentEmail(dependentUser, false);
+    }
+
+    /**
+     * When creating or updating an OTP user, extract a list of newly defined dependents and send an 'accept dependent'
+     * email to each. Then update which dependents have been sent an email so subsequent updates do not trigger
+     * additional emails.
+     */
+    public static void manageAcceptDependentEmail(OtpUser dependentUser, boolean isTest) {
+        if (dependentUser.relatedUsers.isEmpty()) {
+            // No related users defined by dependent.
+            return;
+        }
+
+        dependentUser.relatedUsers
+            .stream()
+            .filter(relatedUser -> relatedUser.acceptKey == null)
+            .forEach(relatedUser -> {
+                String acceptKey = UUID.randomUUID().toString();
+                OtpUser userToReceiveEmail = Persistence.otpUsers.getOneFiltered(eq(EMAIL_FIELD_NAME, relatedUser.email));
+                if (userToReceiveEmail != null && (isTest || sendAcceptDependentEmail(dependentUser, userToReceiveEmail, acceptKey))) {
+                    relatedUser.acceptKey = acceptKey;
+                }
+            });
+
+        // Preserve email sent status (by storing the accept key).
+        Persistence.otpUsers.replace(dependentUser.id, dependentUser);
+    }
+
+    /**
+     * Send 'accept dependent' email.
+     */
+    private static boolean sendAcceptDependentEmail(OtpUser dependentUser, OtpUser relatedUser, String acceptKey) {
+        Locale locale = I18nUtils.getOtpUserLocale(relatedUser);
+
+        String acceptDependentLinkLabel = Message.ACCEPT_DEPENDENT_EMAIL_LINK_TEXT.get(locale);
+        String acceptDependentUrl = getAcceptDependentUrl(acceptKey);
+        String addressee = (dependentUser.name != null) ? dependentUser.name : dependentUser.email;
```

Comment:
Use `Strings.isBlank(dependentUser.name)` (includes null and blank).

Reply:
Updated: https://github.com/ibi-group/otp-middleware/pull/256/commits/99341cb2a25e6633d69fb424482c7936fb6adccc